test/e2e/registration_service_test.go

```diff
@@ -39,15 +44,438 @@ func (s *registrationServiceTestSuite) TestLandingPageReachable() {
 	// just make sure that the landing page is reachable
 	req, err := http.NewRequest("GET", s.route, nil)
 	require.NoError(s.T(), err)
-	client := &http.Client{
+	client := getClient()
+
+	resp, err := client.Do(req)
+	require.NoError(s.T(), err)
+	assert.Equal(s.T(), http.StatusOK, resp.StatusCode)
+}
+
+func (s *registrationServiceTestSuite) TestHealth() {
+	s.Run("verify_healthcheck", func() {
+		// Call health endpoint.
+		req, err := http.NewRequest("GET", s.route+"/api/v1/health", nil)
+		require.NoError(s.T(), err)
+		client := getClient()
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+		assert.Equal(s.T(), http.StatusOK, resp.StatusCode)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.NoError(s.T(), err)
+
+		// Verify JSON response.
+		alive := mp["alive"]
+		require.IsType(s.T(), alive, true)
+		require.True(s.T(), alive.(bool))
+
+		environment := mp["environment"]
+		require.IsType(s.T(), environment, "")
+		require.Equal(s.T(), "e2e-tests", environment.(string))
+
+		revision := mp["revision"]
+		require.NotNil(s.T(), revision)
+
+		buildTime := mp["buildTime"]
+		require.NotNil(s.T(), buildTime)
+
+		startTime := mp["startTime"]
+		require.NotNil(s.T(), startTime)
+	})
+}
+
+func (s *registrationServiceTestSuite) TestAuthConfig() {
+	s.Run("verify_authconfig", func() {
+		// Call authconfig endpoint.
+		req, err := http.NewRequest("GET", s.route+"/api/v1/authconfig", nil)
+		require.NoError(s.T(), err)
+		client := getClient()
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+		assert.Equal(s.T(), http.StatusOK, resp.StatusCode)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
```

Comment:
Reg-service provides auth config as a string. In e2e test env it happens to be a JSON which points to a temporal dev solution (osio prod preview sso). But it's not part of the reg-service API.
So, I would skip verifying the content of this response here. It won't help us catching any issues in prod but will complicate any changes in auth config for dev env (such changes would require updating e2e tests). Let's just make sure that authconfig returns an not empty string.

Reply:
fixed: 7c694e1b8a02e6479a89d0dcc6a08083c2dabed2